apps/obsidian/src/components/canvas/DiscourseNodePanel.tsx

```diff
@@ -0,0 +1,344 @@
+  const nodeTypes = plugin.settings.nodeTypes;
+
+  useEffect(() => {
+    if (!focusedNodeTypeId) return;
+    const exists = !!getNodeTypeById(plugin, focusedNodeTypeId);
+    if (!exists) setFocusedNodeTypeId(null);
+  }, [focusedNodeTypeId, plugin]);
+
+  const focusedNodeType = focusedNodeTypeId
+    ? getNodeTypeById(plugin, focusedNodeTypeId)
+    : null;
+
+  const displayNodeTypes = focusedNodeType ? [focusedNodeType] : nodeTypes;
+
+  useEffect(() => {
+    if (!focusedNodeTypeId) return;
+
+    const handlePointerDown = (e: PointerEvent) => {
+      if (e.defaultPrevented || e.button !== 0) return;
+      const panelEl = rPanelContainer.current;
+      if (!panelEl) return;
+
+      const editorContainer = panelEl.closest(".tldraw__editor");
+      const target = e.target as Element;
+      if (!editorContainer || !target) return;
+      if (!editorContainer.contains(target)) return;
+      if (panelEl.contains(target)) return;
+      if (target.closest("[class^='tlui-']")) return;
+
+      const screenPoint = new Vec(e.clientX, e.clientY);
+      const pagePoint = editor.screenToPage(screenPoint);
+      const nodeType = getNodeTypeById(plugin, focusedNodeTypeId);
+      if (!nodeType) {
+        setFocusedNodeTypeId(null);
+        return;
+      }
+
+      openCreateDiscourseNodeAt({
+        plugin,
+        canvasFile,
+        tldrawEditor: editor,
+        position: pagePoint,
+        initialNodeType: nodeType,
+      });
+
+      setFocusedNodeTypeId(null);
+      e.preventDefault();
+      e.stopPropagation();
+    };
+
+    const handleKeyDown = (e: KeyboardEvent) => {
+      if (e.key === "Escape") {
+        setFocusedNodeTypeId(null);
+      }
+    };
+
+    document.addEventListener("pointerdown", handlePointerDown, true);
+    document.addEventListener("keydown", handleKeyDown);
+    const panelEl = rPanelContainer.current;
+    const editorContainer = panelEl?.closest(
+      ".tldraw__editor",
+    ) as HTMLElement | null;
+    const previousCursor = editorContainer?.style.cursor;
+    if (editorContainer) editorContainer.style.cursor = "copy";
+    return () => {
+      document.removeEventListener("pointerdown", handlePointerDown, true);
+      document.removeEventListener("keydown", handleKeyDown);
+      if (editorContainer) editorContainer.style.cursor = previousCursor || "";
+    };
+  }, [focusedNodeTypeId, editor, plugin, canvasFile]);
+
+  useEffect(() => {
+    const cursor = focusedNodeTypeId ? "cross" : "default";
+    editor.setCursor({ type: cursor });
+    return () => {
+      editor.setCursor({ type: "default" });
+    };
+  }, [focusedNodeTypeId, editor]);
+
+  const handleItemClick = (id: string) => {
+    if (didDragRef.current) return;
+    if (focusedNodeTypeId) {
+      setFocusedNodeTypeId(null);
+      return;
+    }
+    setFocusedNodeTypeId(id);
+  };
+
+  return (
+    <div className="tlui-layout__top__right">
+      <div
+        className="tlui-style-panel tlui-style-panel__wrapper p-2"
```

Comment:
The button `css` is still off compared to Roam's. This actually bother's me quite a bit. I think we should aim for product unity and cohesion wherever possible (first impressions go a long way) and in this instance is should be quite easy (actually I assumed it would be the path of least resistance). Please remove the `p-2`.